signature/policy_eval_sigstore_test.go

```diff
@@ -320,13 +348,18 @@ func TestPRrSigstoreSignedIsSignatureAccepted(t *testing.T) {
 
 	// Successful key+Rekor use
 	pr, err = newPRSigstoreSigned(
-		PRSigstoreSignedWithKeyPath("fixtures/cosign2.pub"),
+		PRSigstoreSignedWithKeyPaths([]string{"fixtures/cosign.pub", "fixtures/cosign2.pub"}),
 		PRSigstoreSignedWithRekorPublicKeyPath("fixtures/rekor.pub"),
 		PRSigstoreSignedWithSignedIdentity(prm),
 	)
+
 	require.NoError(t, err)
 	sar, err = pr.isSignatureAccepted(context.Background(), testKeyRekorImage, testKeyRekorImageSig)
+	require.NoError(t, err)
 	assertAccepted(sar, err)
+	tr, err := pr.prepareTrustRoot()
```

Comment:
I don’t see that this is obviously valid: a new `prepareTrustRoot` is going to load the keys from scratch, so it can set up new pointers for the public keys. `assert.Equal` below then does a `reflect.DeepEqual`, which … is probably fine for confirming equality, but is probably not guaranteed to be true if the crypto had a native implementation.

---

If we want this check, I think it would make sense to make `prSigstoreSigned.isSignatureAccepted` basically:
```go
tr := pr.pepareTrustRoot()
return tr.isSignatureAccepted(…)
```

and then we could run the two parts separately in this test …

---

… but, on the whole, I don’t think the key equality test is actually necessary; if we test that the signatures are correctly accepted/rejected with 0/1/2 keys, that’s fine, and we don’t need a key equality check to re-confirm the logic did the right thing.

Reply:
… actually, there is a defined equality interface https://pkg.go.dev/crypto#PublicKey .

Anyway… I think it might make sense to check that `prepareTrustRoot` loads data correctly, but checking `isSignatureAccepted` can be done just by observing outcomes.